### PR TITLE
Add basic URL validation to UrlField

### PR DESCRIPTION
**Pull request by jschwartz79**

### CODE REVIEW

File: src/springfield/fields.py [hunk 1]
```diff
@@ -339,6 +340,20 @@ class UrlField(StringField):
     """
     :class:`Field` with a URL value
     """
+    def adapt(self, value):
+        """
+        Validate that the `value` has a valid URL format containing
+        a scheme and network location using urlparse.
+
+        Supported schemes include `http`, `https` and `ftp`.
```
Comment (six8) [inline src/springfield/fields.py]: Do we want to restrict schemes? I suppose that's the difference between an URI and URI but there are a lot more schemes possible.

File: src/tests/test_fields.py [hunk 2]
```diff
@@ -26,3 +27,31 @@ def test_float():
...
+    Assure that url performs some basic validation
+    """
+    urlify = fields.UrlField().adapt
+
+    # positive tests
+    for input, expect in [
+        ('http://www.google.com', 'http://www.google.com'),
+        ('hTTp://www.google.com', 'hTTp://www.google.com'),
+        ('ftp://www.google.com', 'ftp://www.google.com'),
+        ('https://www.google.com', 'https://www.google.com')
```
Comment (six8) [inline src/tests/test_fields.py]: add some tests with paths

File: src/tests/test_fields.py [hunk 3]
```diff
@@ -26,3 +27,31 @@ def test_float():
...
+def test_url():
+    """
+    Assure that url performs some basic validation
+    """
+    urlify = fields.UrlField().adapt
+
+    # positive tests
+    for input, expect in [
+        ('http://www.google.com', 'http://www.google.com'),
+        ('hTTp://www.google.com', 'hTTp://www.google.com'),
```
Comment (six8) [inline src/tests/test_fields.py]: I'd prefer that we lowercase scheme and hostname for normalization, but not path/query.

File: src/springfield/fields.py (hunk 1 above)
Comment (jschwartz79) [inline src/springfield/fields.py]: I can just verify we found something for that field if you prefer to leave it more open.

Comment (jschwartz79) [inline src/springfield/fields.py]: We will also miss scheme validation like 'htp' instead of 'http'.

Comment (six8) [inline src/springfield/fields.py]: I think just requiring a scheme is enough. Until we implement more thorough validation we don't want to be too restrictive.

**Comment by six8** [comment]
LGTM.

**Comment by six8** [comment]
@jschwartz79 can you retarget this to develop instead of master?

**Comment by six8** [comment]
See https://github.com/six8/springfield/pull/4 for develop targeted pull request.

**Pull request closed without merging**